conftest.py

```diff
@@ -56,11 +56,16 @@
 
 def pytest_addoption(parser):
     parser.addoption("--runslow", action="store_true", help="run slow tests")
+    parser.addoption("--runarrayexpr", action="store_true", help="run array-expr tests")
 
 
 def pytest_runtest_setup(item):
     if "slow" in item.keywords and not item.config.getoption("--runslow"):
         pytest.skip("need --runslow option to run")
+    if "array_expr" in item.keywords and not item.config.getoption("--runarrayexpr"):
+        pytest.skip("need --runarrayexpr option to run")
+    elif "array_expr" not in item.keywords and item.config.getoption("--runarrayexpr"):
+        pytest.skip("only array-expr tests are being run")
```

Comment:
I'm not very familiar with keywords. What I thought of when mentioning markers is https://docs.pytest.org/en/stable/example/markers.html

you'd annotate individual tests with `@pytest.mark.array_expr` or entire modules with `pytestmark = pytest.mark.array_expr` and could run all tests marked with this by using `pytest -m array_expr` without adding any custom code.

Reply:
Yeah, I tried that but I couldn't get the inverse (not array-expr) to run properly. I'll take another stab at it.

Reply:
I must have made a stupid mistake along the way, things are working properly now locally, lets see if ci works too